test/models/archived_patient_test.rb

```diff
@@ -0,0 +1,90 @@
+require 'test_helper'
+
+class ArchivedPatientTest < ActiveSupport::TestCase
+  before do
+    @user = create :user
+    @patient = create :patient, other_phone: '111-222-3333',
+                                other_contact: 'Yolo'
+
+    @patient.calls.create attributes_for(:call, created_by: @user, status: 'Reached patient')
+    create_language_config
+    @archived_patient = create :archived_patient, line: 'DC',
+                                initial_call_date: 400.days.ago,
+                                created_by_id: @user.id
+  end
+
+  describe 'validations' do
+    it 'should build' do
+      assert @archived_patient.valid?
+    end
+
+    it 'requires a line' do
+      @archived_patient.line = nil
+      refute @archived_patient.valid?
+    end
+
+    it 'requires an initial call date' do
+      @archived_patient.initial_call_date = nil
+      refute @archived_patient.valid?
+    end
+
+    it 'requires a logged creating user' do
+      @archived_patient.created_by_id = nil
+      refute @archived_patient.valid?
+    end
+  end
+
+
+  describe 'Archiving Methods' do
+    before do
+      @patient_old = create :patient, primary_phone: '222-222-3333',
+                                      other_phone: '222-222-4444',
+                                      initial_call_date: 600.days.ago
+      @patient_fulfilled = create :patient, primary_phone: '222-222-3334',
+                                   other_phone: '222-222-4443',
+                                   initial_call_date: 310.days.ago
+      @patient_fulfilled.fulfillment.fulfilled = true
```

Comment:
Might be slightly cleaner to do this with `update`, e.g.:

```
patient.update fulfillment: { fulfilled: true, date_of_check: 300.days.ago, ...}
```

That way you can be sure to get the state of the db in order before the tests start.